packages/plugin-core/src/commands/CopyNoteLink.ts

```diff
@@ -37,8 +45,52 @@ export class CopyNoteLinkCommand extends BasicCommand<
     window.showInformationMessage(`${link} copied`);
   }
 
+  private async getUserLinkAnchorPreference(): Promise<
+    "line" | "block" | null
+  > {
+    const { config, wsRoot } = ExtensionProvider.getDWorkspace();
+    let anchorType = ConfigUtils.getNonNoteLinkAnchorType(config);
+    if (anchorType === undefined) {
+      // The preferred anchor type is not set, so ask the user if they want line numbers or block anchors
+      const preference = await window.showQuickPick(
+        [
+          {
+            label: "block",
+            description: "Use block anchors like `^fx2d`",
+            detail:
+              "Links will always point to the right place, but Dendron must insert a short comment into the file.",
+          },
+          {
+            label: "line",
+            description: "Use line numbers `L123`",
+            detail: "Links may point to the wrong place once code is changed.",
+          },
+        ],
+        {
+          canPickMany: false,
+          ignoreFocusOut: true,
+          title:
+            "What type of anchors should Dendron create for links in non-note files?",
```

Comment:
its not clear from this prompt that this will actually write the preference in `dendron.yml`. In general, we haven't had a precedent for modifying the configuration and I can see this as a valid case as well (`prompt` mode). a proposal:

- set the default mode to `prompt` which will give this prompt on selection
- if the user escapes from prompt, should just copy without any anchors
- do not update the config on user selection

Reply:
> do not update the config on user selection

To confirm, are you saying it should prompt every time until the user goes and modifies the configuration file?

Reply:
yeah. we shouldn't update the user configuration, this should be the behavior if dendron is in prompt mode